PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/configuration_algorithm_classes/completion_construction_algorithms.cpp

```diff
@@ -1,17 +1,42 @@
 #include "completion_construction_algorithms.h"
 
-CompletionConstructionAlgorithms::CompletionConstructionAlgorithms(): dataToBeReadBits(0), registerLengthInDW(0){}
+CompletionConstructionAlgorithms::CompletionConstructionAlgorithms(): dataToBeReadBits(0), registerLengthInBytes(0){}
 
 /**
  * @brief Constructing a Completion TLP with Data Payload inside (return of a Configuration Read request)
  * 
  * @return TLP -> The Constructed TLP (CplD)
  */
+
 TLP CompletionWithData::constructTLP()
 {
```

Comment:
Can we extend it to work properly even with memory requests?

Reply:
I think we will also need to pass the TLP as a parameter to the function, as I am saying I want a completion for this one specifically.

Reply:
And for the second comment, indeed I have already made what you are saying and you can find this is the pull request(there is a function called setTLP() inside the TlpConstructor class to do so), it's just un-merged yet.